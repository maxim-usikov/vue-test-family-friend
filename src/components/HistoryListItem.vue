<template>
  <li>
    {{ id }}.
    ({{ createdAt | hhmm }})
    {{ query }}
    <span :class="$style.RepositoryCount">{{ repositoryCount }}</span>
  </li>
</template>

<script>
import hhmm from '@/utils/date.hhmm';

export default {
  name: 'HistoryListItem',

  filters: {
    hhmm,
  },

  props: {
    id: {
      type: Number,
      required: true,
    },
    query: {
      type: String,
      required: true,
    },
    createdAt: {
      type: Date,
      required: true,
    },
    response: {
      type: Object,
      required: true,
    },
  },

  computed: {
    repositoryCount() {
      return this.response.data.search.repositoryCount.toLocaleString();
    },
  },
};
</script>

<style module>
.RepositoryCount {
  float: right;
}
</style>
