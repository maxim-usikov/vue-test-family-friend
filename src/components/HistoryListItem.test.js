import { shallowMount } from '@vue/test-utils';

import HistoryListItem from './HistoryListItem.vue';

describe('HistoryListItem', () => {
  let $c;

  beforeEach(() => {
    $c = shallowMount(HistoryListItem, {
      propsData: {
        id: 777,
        query: 'test',
        createdAt: new Date('December 26, 1990 20:00:00'),
        response: { data: { search: { repositoryCount: 999 } } },
      },
    });
  });

  it('is a Vue instance', () => {
    expect($c.isVueInstance()).toBeTruthy();
  });

  it('name is `HistoryListItem`', () => {
    expect($c.name()).toBe('HistoryListItem');
  });

  it('renders correctly', () => {
    expect($c.element).toMatchSnapshot();
  });
});
