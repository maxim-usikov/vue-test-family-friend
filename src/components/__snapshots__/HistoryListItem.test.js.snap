// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HistoryListItem renders correctly 1`] = `
<li>
  
  777.
  (20:00)
  test
  
  <span
    class="RepositoryCount"
  >
    999
  </span>
</li>
`;
